lib/temp-directory.js

```diff
@@ -9,14 +9,17 @@ var osenv = require('osenv');
 
 function create(context, next) {
   var _path = path.join(osenv.tmpdir(),uuid.v4());
-  var npmConfigTmp = path.join(_path, 'npm_config_tmp');
 
-  context.path = _path;
-  context.npmConfigTmp = npmConfigTmp;
+  if (context.options && context.options.tmpDir) {
+    context.path = path.join(context.options.tmpDir, context.module.name);
```

Comment:
this line should still use uuid to generate a nonce. 

I think we can likely get rid of _path and make the path in line